src/Ng/Jobs/EndpointMonitoringJob.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NuGet.Services.Configuration;
+using NuGet.Services.Metadata.Catalog;
+using NuGet.Services.Metadata.Catalog.Monitoring;
+using NuGet.Protocol.Core.Types;
+using NuGet.Protocol;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test;
+using System.Linq;
+using NuGet.Services.Metadata.Catalog.Monitoring.Resources;
+using NuGet.Services.Metadata.Catalog.Monitoring.Providers;
+using NuGet.Configuration;
+
+namespace Ng.Jobs
+{
+    public class EndpointMonitoringJob : LoopingNgJob
+    {
+        private CommitCollector _collector;
+        private ReadWriteCursor _front;
+        private ReadCursor _back;
+
+        public EndpointMonitoringJob(ILoggerFactory loggerFactory) : base(loggerFactory)
+        {
+        }
+
+        public override string GetUsage()
+        {
+            return "Usage: ng endpointmonitoring "
+                   + $"-{Arguments.Gallery} <v2-feed-address> "
+                   + $"-{Arguments.Source} <catalog> "
+                   + $"-{Arguments.StorageBaseAddress} <storage-base-address> "
+                   + $"-{Arguments.StorageType} file|azure "
+                   + $"[-{Arguments.StoragePath} <path>]"
+                   + "|"
+                   + $"[-{Arguments.StorageAccountName} <azure-acc> "
+                   + $"-{Arguments.StorageKeyValue} <azure-key> "
+                   + $"-{Arguments.StorageContainer} <azure-container> "
+                   + $"-{Arguments.StoragePath} <path> "
+                   + $"[-{Arguments.VaultName} <keyvault-name> "
+                   + $"-{Arguments.ClientId} <keyvault-client-id> "
+                   + $"-{Arguments.CertificateThumbprint} <keyvault-certificate-thumbprint> "
+                   + $"[-{Arguments.ValidateCertificate} true|false]]] "
+                   + $"-{Arguments.StorageTypeAuditing} file|azure "
+                   + $"[-{Arguments.StoragePathAuditing} <path>]"
```

Comment:
Do we need to support  Arguments.StorageType either? Have we even tried this thing against the file system?